### PR TITLE
Fixes for new TOPPView

### PR DESCRIPTION
**Pull request by cbielow**
## Description

Fixes some issues mentioned in https://github.com/OpenMS/OpenMS/issues/6423, probably caused by Refactoring PR #6139.

Notably:
 + fix plotting of precursor positions in 2D View
 + show intensity when mouse-hovering over a data point
image
 + position of annotation items (e.g. for MS2 b/y ions)
 + 'Save IDs' does not throw anymore
 + panning glitches fixed
 + intensity mode in 1D view fixed (log, percentage, normal etc)

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Review by timosachsenberg — APPROVED** [review]

**Comment by poshul** [comment]
rebuild jenkins

**Comment by jpfeuffer** [comment]
ALL of them actually show this

**Review by jpfeuffer — CHANGES_REQUESTED** [review]
see comment above

**Comment by jpfeuffer** [comment]
When it zooms it switches axes or some weird stuff

**Comment by timosachsenberg** [comment]
MS2 look better for me now but I also observe the switching when zooming or panning:
image
just panned a bit and looks good again:
image

**Comment by jpfeuffer** [comment]
With MS2 I meant the 1D view.
Nothing is shown there anymore

**Comment by timosachsenberg** [comment]
works for me (there was just a scaling to 1.0 issue that went away after pressing one of the scale buttons)
image

**Comment by timosachsenberg** [comment]
zooming in by mouse wheel seems to work but selecting a region will not automatically scale:
image
only if % button is pressed

**Comment by timosachsenberg** [comment]
```
diff --git a/src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp b/src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp
@@ -1915,6 +1919,8 @@ namespace OpenMS
     }
     selection_view_->setCurrentIndex(DataSelectionTabs::IDENT_IDX); //switch to ID view
     selection_view_->currentTabChanged(DataSelectionTabs::IDENT_IDX);
+    setIntensityMode(PlotCanvas::IM_SNAP);
+    updateMenu();
   }
```
in annotateWithIdentifications seem to fix the first issue of initial scaling to 1.0

**Comment by timosachsenberg** [comment]
regarding the peak annotations I found this version that has the improved logic: 

https://github.com/OpenMS/OpenMS/blob/973674372b1e97e542acdb6d59295f492b3c051c/src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DPeakItem.cpp

File: src/openms_gui/source/VISUAL/Plot2DCanvas.cpp [hunk 1]
```diff
@@ -606,6 +606,14 @@ namespace OpenMS
     QStringList lines;
     lines << unit_mapper_.getDim(DIM::X).formattedValue(xy_point.getX()).toQString();
     lines << unit_mapper_.getDim(DIM::Y).formattedValue(xy_point.getY()).toQString();
+    if (unit_mapper_.getDim(DIM::X).getUnit() != DIM_UNIT::INT && unit_mapper_.getDim(DIM::X).getUnit() != DIM_UNIT::INT)
```
Comment (timosachsenberg) [inline src/openms_gui/source/VISUAL/Plot2DCanvas.cpp:609]:
```suggestion
    if (unit_mapper_.getDim(DIM::X).getUnit() != DIM_UNIT::INT && unit_mapper_.getDim(DIM::Y).getUnit() != DIM_UNIT::INT)
```

**Comment by cbielow** [comment]
> MS2 look better for me now but I also observe the switching when zooming or panning: image just panned a bit and looks good again: image

this was caused by trying to auto-guess the MS level when creating an AreaIterator. Bad idea.
Its fixed now.

**Comment by cbielow** [comment]
Pressing `Save IDs` is now also fixed, with some very interesting background story...
https://github.com/OpenMS/OpenMS/pull/6519/commits/83580fb35e65a12d062a3b40c031da84fe4d04a2

**Comment by timosachsenberg** [comment]
nice. another thing:
I think there might be something wrong with mapping the intensity type to the button index in the button group.
This could be some internal change in Qt but maybe it is not safe to translate the index of the button group directly to the intensity mode.
image

**Comment by timosachsenberg** [comment]
image

File: src/openms/include/OpenMS/KERNEL/AreaIterator.h [hunk 2]
```diff
@@ -79,18 +79,10 @@ namespace OpenMS
...
-          // if begin is dereferencable ...
-          if (begin != end)
-          {
-            ms_level_ = begin->getMSLevel();
-          }
-          else
-          {
-            ms_level_ = 1;
-          }
+          ms_level_ = ms_level;
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/KERNEL/AreaIterator.h:85]: move to initializer list?

Comment (cbielow) [reply src/openms/include/OpenMS/KERNEL/AreaIterator.h:85]: yes :)

**Review by timosachsenberg — APPROVED** [review]
only minor / optional things

**Comment by cbielow** [comment]
now, also the position of annotations is fixed.
This should fix all things related to the large TOPPView pr.
Ready for merge (if tests pass).

**Review by jpfeuffer — APPROVED** [review]

**Pull request merged**